### ISSUE TITLE
The keyboard navigation breaks when nested menus aren't defined in the same template as the parent menu

### ISSUE THREAD

**Issue by asenradev** [vmware/clarity#5329]
## Describe the bug
**My requirement**
I have a use case to show dropdown/menu from a dynamic definition. Just imagine that we request a hierarchical structure from the server and should be rendered as dropdown.

**Approach**
I came up with an implementation relying on recursive instantiation of <ng-template>. Find a very simplified demonstration of the approach in the StackBlitz

**Problem**
Everything works fine out-of-the-box except the keyboard navigation. Expanding nested menus and moving the focus to their children for example is not working.

## How to reproduce

https://stackblitz.com/edit/clarity-v4-dropdown-template-keyboard-navigation

1. move focus to a nested menu trigger item like 'menu1'
2. expand it with arrow left

## Expected behavior

The focus is moved to the first item in the nested clr-dropdown-menu.

## Additional notes

Basically all the keyboard navigation event that rely on the parent-child relation between menus are not handled.

I have a workaround that works fine to my perceptions but it touches internals. I will be glad to discuss it with core team member so at least I improve my workaround to a level that's not extremely fragile.

**Comment by bbogdanov** [comment]
Hi, As we are moving our focus to `@clr/core` development, we are less likely to introduce changes in `@clr/angular` package.

Please share your workaround so we can give you some directions.

**Comment by asenradev** [comment]
Here is the workaround: https://stackblitz.com/edit/clarity-v4-dropdown-template-keyboard-navigation-workaround-1?file=src%2Fapp%2Ffocus%2F

Summary of the changes:
 - FocusableItemCollector does the same thing as ClrDropdownMenu - query the focusable child items and add them to the parent DropdownFocusHandler
- FocusableMenu and FocusableDropdownItem are providing the focusable items. If DropdownFocusHandler and FocusableItem were public, I wouldn't need these directives and touch the private fields of ClrDropdownMenu and ClrDropdownItem 
- the parent menu is passed to the nested template via `[ngTemplateOutletContext]="{ $implicit: menuItem.children, parentMenu: nestedDropdownMenu }">`

**Comment by gnomeontherun** [comment]
There is a workaround available for this issue, and we recommend using it for Clarity Angular. We would consider a contribution to address this issue if someone is able to work out a solution. As we build Clarity Core implementations, we expect that this issue won’t be an issue or will be configurable on the application side. To help us clean up our backlog, we are going to close this with a functional workaround available and suggest you follow updates for Clarity Core for enhancements that can support your use case with Clarity Core components.
